iroh-bytes/src/util.rs

```diff
@@ -256,6 +256,14 @@ impl FromStr for Hash {
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         let sb = s.as_bytes();
+        if sb.len() == 64 {
```

Comment:
does b3sum pad?

Reply:
what would there be to pad? a 32 byte hash is always 64 hex chars, or not?

```
❯ head -c 10 /dev/urandom | b3sum
e93842b50651bacd7b51beed6f4549ae6cdba26e657c11c1874e145f4af41055  -
```

Reply:
Since we allow all multiformats, I guess there could be some multiformat that is 64 byte long. But I think it is pretty unlikely (impossible?) to have a multiformat that is 64 byte long and all hex chars.

Reply:
> 64 hex chars, or not 

ah yes I think so, it was base64 with its many variants of padding

Reply:
that's exactly why hex is so nice...